patches/tModLoader/Terraria/ModLoader/Default/Developer/Jofairden/AndromedonEffect.cs

```diff
@@ -70,54 +70,43 @@ internal sealed class AndromedonEffect : ICloneable
 			_lastLife = player.statLife;
 		}
 
-		public void ModifyDrawLayers(Mod mod, Player player, List<PlayerLayer> layers) {
+		public IEnumerable<PlayerDrawLayer> AddDrawLayers(Mod mod, Player player) {
+			if (LayerStrength < 0f) {
+				yield break;
+			}
+
 			_headSlot ??= mod.GetEquipSlot($"PowerRanger_{EquipType.Head}", EquipType.Head);
 			_bodySlot ??= mod.GetEquipSlot($"PowerRanger_{EquipType.Body}", EquipType.Body);
 			_legSlot ??= mod.GetEquipSlot($"PowerRanger_{EquipType.Legs}", EquipType.Legs);
 
-			if (LayerStrength >= 0f) {
-				int i;
+			static PlayerDrawLayer Layer(PlayerDrawLayer layer, float depth) {
```

Comment:
~~definitely hate this, surely there's a way to not have to reset visible every frame, or modify depth.~~
Shouldn't the Andromedon draw layers just be changed to extend `PlayerDrawLayer` and set their visible flags themselves?

Reply:
Ye, I have an outdated version of that in a stash, will update it.